src/theme/colorScheme.ts

```diff
@@ -0,0 +1,71 @@
+export enum ColorScheme {
+  Dark = "Dark",
+  Light = "Light",
+  System = "System",
+}
+
+// Get a single reference to a `MediaQueryList` that the app can use to (a) determine
+// whether the user's system is configured to prefer a dark color scheme and (b) to
+// potentially react to changes in that configuration.
+//
+// References:
+// - https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList
+// - https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme
+//
+export const documentMQL: MediaQueryList = window.matchMedia(
+  "(prefers-color-scheme: dark)",
+);
+
+/**
+ * Returns `true` if the value is a valid color scheme.
+ */
+export const isValidColorScheme = (value: unknown) => {
+  return Object.values(ColorScheme).includes(value as ColorScheme);
+};
+
+/**
+ * Toggles the dark color scheme according to the flag passed in.
+ *
+ * Note: Activating/deactivating the dark color scheme is a matter of adding/removing
+ *       the `.ion-palette-dark` class from the `<html>` element.
+ *
+ * Reference: https://ionicframework.com/docs/theming/dark-mode#css-class
+ *
+ * @param wantsDarkColorScheme {boolean} Whether you want the dark color scheme to be enabled.
+ */
+export const toggleDarkColorScheme = (wantsDarkColorScheme: boolean) => {
+  const htmlEl = document.documentElement;
```

Comment:
Selecting this on each call is unnecessary (although I don't think there is any significant performance penalty). Consider defining this one time, near the top of this module, outside of any functions.